esp/bindings/SOAP/xpp/xpp/XmlPullParser.h

```diff
@@ -174,7 +174,8 @@ namespace xpp {
     }
 
 
-    int skipSubTree()  {
+    bool skipSubTree()  {
```

Comment:
The platform isn't currently using the return value, so this change looks reasonable. But should we assume that nothing else uses the existing return value?

Reply:
The code couldn't make any real use of it since it always returned XmlPullParser::END_TAG whether it cycled through some children or not, but I'll make a new function anyway.

Reply:
Your change is fine to address the original comment, but I am concerned about the original implementation.

If skipSubTree is used on a document not known to be well-formed, an unbalanced start tag can put it into an infinite loop. It only leaves the loop when the end tags balance. I remember writing code before where I checked to make sure I wasn't at the end of the document after skipping. I didn't realize until now that it's an unhandled error, unless I've misread next() and it doesn't return END_DOCUMENT infinitely.

It's also not clear that the well-formed behavior is consistent given different starting positions. Skipping from the start tag skips the entire element. Skipping from content skips the remainder of the current element. Skipping from the end tag of the current element skips the remainder of the parent element (and foolishly skipping from the end of the root element will never return). The first two make sense to me, but I could easily see the third being a NO-OP instead.